android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java

```diff
@@ -27,22 +27,33 @@ public static boolean isActive() {
 
     public static boolean prevent = false;
 
-    public static void sendEvent(boolean up) {
+    public static void sendEvent(boolean up, int amount) {
         if (active) {
             WritableMap args = Arguments.createMap();
+            args.putInt("scrollAmount", amount); // add scroll amount to event data
             if (up)
                 appContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit("VolumeUp", args);
             else
                 appContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit("VolumeDown", args);
         }
     }
 
-    public static void down() {
-        sendEvent(false);
+    public static void down(int amount) {
+        sendEvent(false, amount);
+    }
+
+    public static void up(int amount) {
+        sendEvent(true, amount);
+    }
+
+    private static int scrollAmount = 100;
+
+    public static void setScrollAmount(int amount) {
+        scrollAmount = amount;
     }
 
-    public static void up() {
-        sendEvent(true);
+    public static int getScrollAmount() {
+        return scrollAmount;
```

Comment:
Why we have to send event with amount value

Reply:
the parameter "amount" is the value representing the amount of scrolling desired when either the volume up or volume down button is pressed.

in the MainActivity class, the dispatchKeyEvent() method listens for volume button presses (up and down) and calls the respective VolumeButtonListener.up() and VolumeButtonListener.down() methods with the scroll amount obtained from VolumeButtonListener.getScrollAmount()

Reply:
Actually the value of `scrollAmout` is stored and persisted by `Redux` (when you call dispatch to settingReducer).
You could remove old eventListener and add the new one when `scrollAmout` get changed.
No need to modify the naitve Module. Anyway I think your logic is overhandling
![image](https://user-images.githubusercontent.com/86464880/229106731-3246d638-a220-4282-9387-7d31a428aafe.png)

Reply:
Thanks, I will look into it. 
Never worked before with Redux/JavaScript, gotta use more chadGPT for knowledge gaps😂👌

Reply:
i pushed a new commit, not sure if this was the correct way(im coming from gerrit, is this like a new patchset?).
I think its cleaner now, now working with redux for all the changes.

I tried to use **KeyboardAvoidingView**, but didnt manage to implement it, would be usefull for the future.

thank you for patience😁

Reply:
The native module is ONLY to notify ReactNative that the button has been pressed. The rest of the behaviour SHOULD BE IN REACT NATIVE not in java...